exporter/jaegerexporter/go.mod

```diff
@@ -23,7 +23,6 @@ require (
 )
 
 require (
-	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 // indirect
```

Comment:
This broke the jaegerexporter: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6027243374/job/16352630002?pr=26314

Reply:
@bryan-aguilar it looks like you've seen this issue before: https://github.com/aws-observability/aws-otel-collector/issues/926.

Reply:
Looks like a few other contrib modules have solved this using a replace statement: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/routingconnector/go.mod#L60.

Reply:
The answer is usually to insert a replace statement. The cause for us is that a tags underlying commit was changes and causes an ambiguous import.

Reply:
Friendly reminder to not change a tags commit after publish. At least for go packages.

Reply:
> The answer is usually to insert a replace statement. The cause for us is that a tags underlying commit was changes after initial publish and that causes an ambiguous import.

Which tags were changed after publishing? cloud.google.com/go/compute/metadata?

Reply:
It's been a while since that import error but I believe it was `cloud.google.com/go`. Note the replace statement that I mentioned [here](https://github.com/aws-observability/aws-otel-collector/issues/926#issuecomment-1263065587) isn't a best practice as it's missing a version on the left hand side. 

We have faced this error with different dependencies. Troubleshooting for it involves look at the upstream dependency to see what the commit attached to the tag is. If it is different than the one posted in `go.pkg` then we make the assumption that the tag was changed after publish. We would then see if a tag that we can insert a replace statement to that has the same functionality. Sometimes we open an issue upstream to see if the package maintainer can fix it.

Reply:
I'm sorry, I think my above statements are wrong. When an underlying commit is changed for a tag I think you get a `checksum mismatch` when running go mod tidy. The ambiguous import error is separate. I mixed the two issues up.